GCP effective firewall listing

jonathan-greig opened the pull request:
Relates to #339

giovannt0 commented on tests/providers/gcp/gcp_mocks.py:
```diff
@@ -556,3 +556,97 @@
+                }
+        ],
+        'fingerprint': 'bm9mcGZwZnA=',
+        'kind': 'compute#networkInterface'}
```
Nit: put the bracket on the next line so we clearly see where the dict finishes

giovannt0 commented on libcloudforensics/providers/gcp/internal/compute.py:
```diff
@@ -1224,6 +1224,85 @@ def DetachServiceAccount(self) -> None:
+      request = gce_instance_client.getEffectiveFirewalls(
+          project=self.project_id, instance=self.name, zone=self.zone,
+          networkInterface=nic_name)
+      response = request.execute()
```
You may wanna wrap this using `common.ExecuteRequest`, so that errors are caught.

jonathan-greig commented:
Addressed code review comments and also did some further normalisation of the fw rules as there were inconsistencies in FW vs FW policy field names `IPProtocol` vs `ipProtocol`

giovannt0 commented on libcloudforensics/providers/gcp/internal/compute.py:
```diff
@@ -1224,6 +1226,112 @@ def DetachServiceAccount(self) -> None:
+      nic_fw_rules = []
+      responses = common.ExecuteRequest(
+          gce_instance_client, 'getEffectiveFirewalls',
+          {'project': self.project_id, 'instance': self.name,
```
Can you make this a separate variable? `request = {...}`

giovannt0 commented on libcloudforensics/providers/gcp/internal/compute.py:
```diff
@@ -1224,6 +1226,112 @@ def DetachServiceAccount(self) -> None:
+      if 'ipProtocol' in config:
+        normalised['ip_protocol'] = config['ipProtocol']
+      else:
+        normalised['ip_protocol'] = config['IPProtocol']
```
If `ipProtocol` is not in `config`, is it always guaranteed that `IPProtocol` is?

jonathan-greig replied:
Yeah according to the documentation each "L4config" will have either an `ipProtocol` if it is a firewall policy or `IPProtocol` if it is a standard firewall rule. Not sure why they are named differently :smiley: . `elif` is probably more suitable here though will change.
